packages/sit-onyx/src/components/OnyxListbox/types.ts

```diff
@@ -17,6 +17,17 @@ export type OnyxListboxProps<TValue extends SelectionOptionValue = SelectionOpti
    * Current value / selected options.
    */
   modelValue?: TValue;
+  /**
+   * Whether to show a loading indicator.
+   */
+  loading?: boolean;
+  /**
+   * How to emit the `loadMore` event:
+   * - lazy: User scrolls to the end of the options
+   * - button: User clicks the load more button. If there are no more options to load,
+   *           set this property to `undefined` so the button is hidden accordingly.
+   */
+  loadingMode?: ListboxLoadingMode;
```

Comment:
the convention that @JoCa96 suggested to me in the OnyxSelect was to offer options that are bound to booleans as config object so they are capsulated neatly and easy to extend in the future. this approach might apply here too.

```suggestion
  /**
   * Whether to show a loading indicator.
   */
  loading?: boolean | {
    /**
     * How to emit the `loadMore` event:
     * - lazy: User scrolls to the end of the options
     * - button: User clicks the load more button. If there are no more options to load,
     *           set this property to `undefined` so the button is hidden accordingly.
     */
    loadingMode?: ListboxLoadingMode;
  }
```

Reply:
just saw the ListboxLoadingMode type definition, this would make sense to be integrated in this config object directly

Reply:
sorry I just realized that loading has nothing to do with loadingMode. loading is about the whole thing. sorry ^^

Reply:
maybe this renaming might make it a bit more intuitively clear :)

```suggestion
  /**
   * Whether to show a loading indicator.
   */
  loading?: boolean;
  /**
   * How to emit the `loadMore` event:
   * - lazy: User scrolls to the end of the options
   * - button: User clicks the load more button. If there are no more options to load,
   *           set this property to `undefined` so the button is hidden accordingly.
   */
  loadMoreMode?: ListboxLoadingMode;
```

Reply:
reviewing further, now I'm totally confused - they __are__ linked, because setting the mode alone is not actually putting it in loading state 🙈

maybe we can have a short call about this to come to a decision ^^